flatdata-cpp/test/BackwardCompatibilityTest.cpp

```diff
@@ -52,22 +51,20 @@ compare_byte_arrays( const Array& expected,
                      const flatdata::MemoryDescriptor actual,
                      const Storage& storage )
 {
-    EXPECT_EQ( expected.size( ) - 1, actual.size_in_bytes( ) )
-        << "Sizes differ. Hexdump: " << std::endl
-        << storage.hexdump( );
+    INFO( "Sizes differ. Hexdump: \n" << storage.hexdump( ) );
+    CHECK( actual.size_in_bytes( ) + 1 == expected.size( ) );
 
     for ( size_t i = 0; i < actual.size_in_bytes( ); ++i )
     {
-        EXPECT_EQ( static_cast< uint8_t >( expected[ i ] ), actual.data( )[ i ] )
-            << "Difference at position " << i << ". Hexdump: " << std::endl
-            << storage.hexdump( );
+        INFO( "Difference at position " << i << ". Hexdump: \n" << storage.hexdump( ) );
+        CHECK( actual.data( )[ i ] == static_cast< uint8_t >( expected[ i ] ) );
     }
 }
 
 void
 fill_signed_struct( SignedStructMutator s )
 {
-    EXPECT_EQ( size_t( 10 ), s.size_in_bytes( ) );
+    REQUIRE( s.size_in_bytes( ) == size_t( 10 ) );
```

Comment:
I guess this will not output both values? Does catch have something that would in case of error? I often find this very convenient.

Reply:
Actually it seems it does:
```
Example.cpp:9: FAILED:
  REQUIRE( Factorial(0) == 1 )
with expansion:
  0 == 1
```

Nice macro/template magic :+1: